lua/nvim-treesitter-textobjects/shared.lua

```diff
@@ -82,35 +80,31 @@ local get_query_matches = memoize(function(bufnr, query_group, root, root_lang)
       local prepared_match = {}
 
       -- Extract capture names from each match
-      for id, node in pairs(match) do
+      for id, nodes in pairs(match) do
         local query_name = query.captures[id] -- name of the capture in the query
+        if type(nodes) ~= "table" then
+          nodes = { nodes }
+        end
         if query_name ~= nil then
           local path = vim.split(query_name, "%.")
-          local range = Range:from_node(node, bufnr)
-          range.metadata = metadata[id]
-          insert_to_path(prepared_match, path, range)
+          for _, node in ipairs(nodes) do
+            insert_to_path(prepared_match, path, { node:range(true) })
+          end
         end
       end
 
       if metadata.range and metadata.range[7] then
         ---@cast metadata TSTextObjects.Metadata
         local query_name = metadata.range[7]
         local path = vim.split(query_name, "%.")
-        insert_to_path(
-          prepared_match,
-          path,
-          Range:new(
-            metadata.range[1],
-            metadata.range[2],
-            metadata.range[3],
-            metadata.range[4],
-            metadata.range[5],
-            metadata.range[6],
-            "-1",
-            "-1",
-            bufnr
```

Comment:
This undoes a change I made deliberately to fix failing tests. Can you comment on why you did that?

Reply:
The last three parameters are for `id`, `parent_id`, and `bufnr`, all of these I removed because these fields only exist on the former `TSTextObject.Range`(In other words, we must remove them). IIRC except for checking whether the two ranges are equivalent, The only usages of these fields in `swap.lua`, interestingly these usages are in a true condition, e.g. `current_range.id == current_range.id`, so I delete them directly. There might be a regression here, but the regression has already happened before this PR.

Reply:
Well, there was a regression that I fixed by re-adding them. Can you confirm that swapping, e.g., function arguments in Lua still works after this change?

Reply:
Okay...I found it could not work, but only because of a typo not related to this, I've fixed it and force-pushed.

https://github.com/nvim-treesitter/nvim-treesitter-textobjects/compare/734f9acc449ea3819a80c1ca32140c9a05bab53b..af0aad9c5498a1a7cbf7b2b5644ab3fad9b5c44e

Reply:
Yeah, I'm afraid the test suite is not comprehensive enough; I had to do a lot of manual testing to find and fix regressions in my last big refactor PR. (Another thing that is not sufficiently covered and needs to be tested manually is using locals for selection; I use the `minimal_init` config and just go through the list of blocks, testing one command from each.)